diff --git a/node_modules/@or13/did-jwk/src/index.js b/node_modules/@or13/did-jwk/src/index.js
index 37bb2c8..4a29eac 100644
--- a/node_modules/@or13/did-jwk/src/index.js
+++ b/node_modules/@or13/did-jwk/src/index.js
@@ -161,7 +161,7 @@ const toDidDocument = (jwk) => {
   const didDocument = {
     '@context': [
       'https://www.w3.org/ns/did/v1',
-      {'@vocab': 'https://www.iana.org/assignments/jose#'},
+      'https://w3id.org/security/suites/jws-2020/v1'
     ],
     'id': did,
     'verificationMethod': [vm],
@@ -186,6 +186,10 @@ const resolve = (did) => {
   const decoded = jose.base64url.decode(methodSpecificId);
   const message = new TextDecoder().decode(decoded);
   const jwk = JSON.parse(message);
+  const {publicKeyJwk} = jwk;
+  if (publicKeyJwk) {
+    return toDidDocument(publicKeyJwk);
+  }
   return toDidDocument(jwk);
 };
 
@@ -274,9 +278,10 @@ class DIDMethodClient {
 
     this.operations = {
       create: async (alg, options = {}) => {
-        const {privateKeyJwk} = await generateKeyPair(alg);
-        const didDocument = toDidDocument(privateKeyJwk);
-        return {privateKeyJwk, didDocument};
+        const keyPair = await generateKeyPair(alg);
+        const {publicKeyJwk} = keyPair;
+        const didDocument = toDidDocument(publicKeyJwk);
+        return {keyPair, didDocument};
       },
       resolve: async (did) => {
         return resolve(did);
